Testing/dotNetRDF.MockServerTests/SparqlServiceTests.cs

```diff
@@ -0,0 +1,223 @@
+            var serviceResults = XmlFormat(CreateResults(numberOfTriples: 2));
+            RegisterSelectQueryGetHandler(QueryWithInlineData(2), serviceResults);
+
+            var results = ProcessQuery(dataset, query);
+
+            results.Should().NotBeNull().And.HaveCount(2);
+            var sparqlLogEntries = _server.FindLogEntries(new RequestMessagePathMatcher(MatchBehaviour.AcceptOnMatch, "/sparql")).ToList();
+            sparqlLogEntries.Should().HaveCount(1);
+            sparqlLogEntries[0].RequestMessage.Method.Should().BeEquivalentTo("get");
+        }
+
+        [Fact]
+        public void ItCallsServiceWithInlineDataInChunks()
+        {
+            var dataset = CreateDataset(numberOfTriples: 102);
+            var query = $"SELECT * WHERE {{ ?s ?p ?o SERVICE <{ServiceUri}> {{?s ?p ?o}} }}";
+            var resultsOfPost = XmlFormat(CreateResults(numberOfTriples: 100)); 
+            var resultsOfGet = XmlFormat(CreateResults(numberOfTriples: 2));
+            RegisterSelectQueryPostHandler(QueryWithInlineData(100), resultsOfPost);
+            RegisterSelectQueryGetHandler(QueryWithInlineData(2), resultsOfGet);
+            
+            var results = ProcessQuery(dataset, query);
+
+            results.Should().NotBeNull().And.HaveCount(102);
+            var sparqlLogEntries = _server.FindLogEntries(new RequestMessagePathMatcher(MatchBehaviour.AcceptOnMatch, "/sparql")).ToList();
+            sparqlLogEntries.Should().HaveCount(2);
+            sparqlLogEntries[0].RequestMessage.Method.Should().BeEquivalentTo("post");
+            sparqlLogEntries[1].RequestMessage.Method.Should().BeEquivalentTo("get");
+        }
+
+        [Fact]
+        public void ItResolvesEndpointFromVariable()
+        {
+            var dataset = CreateDataset(numberOfTriples: 1);
+            var query = $"SELECT * WHERE {{ VALUES ?service {{<{ServiceUri}>}} SERVICE ?service {{?s ?p ?o}} }}";
+            var serviceResults = XmlFormat(CreateResults(numberOfTriples: 1));
+            RegisterSelectQueryGetHandler(SameAsQuery("SELECT * WHERE {?s ?p ?o}"), serviceResults);
+
+            var results = ProcessQuery(dataset, query);
+
+            results.Should().NotBeNull().And.HaveCount(1);
+            var sparqlLogEntries = _server.FindLogEntries(new RequestMessagePathMatcher(MatchBehaviour.AcceptOnMatch, "/sparql")).ToList();
+            sparqlLogEntries.Should().HaveCount(1);
+            sparqlLogEntries[0].RequestMessage.Method.Should().BeEquivalentTo("get");
+        }
+
+        [Fact]
+        public void ItResolvesNoEndpointFromVariable()
+        {
+            var dataset = CreateDataset(numberOfTriples: 0);
+            var query = $"SELECT * WHERE {{ VALUES ?service {{ 'noturi' }} SERVICE ?service {{?s ?p ?o}} }}";
+
+            var results = ProcessQuery(dataset, query);
+
+            results.Should().NotBeNull().And.HaveCount(0);
+        }
+
```

Comment:
That's interesting! This matches the logic that we had for SERVICE previously so I'm not complaining, but I wonder if this behaviour is slightly at odds with the specification as https://www.w3.org/TR/2013/REC-sparql11-federated-query-20130321/#serviceFailure says that a query should fail if the endpoint IRI cannot be resolved (though it doesn't say what to do if the SERVICE endpoint value is not an IRI).

In any case I'm happy to keep the behaviour we had until someone complains :-)